src/Workspaces/Core/Portable/Diagnostics/DiagnosticDataLocation.cs

```diff
@@ -124,7 +130,7 @@ internal FileLinePositionSpan GetFileLinePositionSpan()
         internal string? GetFilePath()
             => GetFilePath(OriginalFilePath, MappedFilePath);
 
-        private static string? GetFilePath(string? original, string? mapped)
+        private static string? GetFilePath(string original, string? mapped)
         {
             if (RoslynString.IsNullOrEmpty(mapped))
```

Comment:
more of a general comment than one on this PR.  This method is weird... it combines the directory name of the original file path with the mapped file path?  Is the mapped file path just a file name?

Reply:
that is correct.  and the naming here is def weird and confusing.

Reply:
this gets better with my oiher PR (that changes these all into FileLinePositionSpans.).